aspnetcore/security/ip-safelist.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Client IP safelist for ASP.NET Core
+author: damienbod
+description: Learn how to write Middleware or action filters to validate remote IP addresses against a list of approved IP addresses.
+ms.author: tdykstra
+ms.custom: mvc
+ms.date: 08/31/2018
+uid: security/ip-safelist
+---
+# Client IP safelist for ASP.NET Core
+
+By [Damien Bowden](https://twitter.com/damien_bod) and [Tom Dykstra](https://github.com/tdykstra)
+ 
+This article shows two ways to implement an IP safelist (also known as a whitelist):
+
+* By using ASP.NET Core middleware to check the remote IP address of every request.
+* By using ASP.NET Core action filters to check the remote IP address of requests for specific action methods.
+
+The sample app illustrates both approaches. In each case, a string containing approved client IP addresses is stored in an app setting. The middleware or filter parses the string into a list and  checks if the remote IP is in the list. If not, an HTTP 403 Forbidden status code is returned.
```

Comment:
>stored in an app setting.

That's an implementation detail they don't need to know about now. A real app would fetch it from a DB or a Web API call. It sounds clumsy removing tho so maybe leave it.

Use VSC DocFX addin to find/remove double spaces. 

an HTTP 403 (Forbidden) status code is returned.